demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -1757,11 +1796,64 @@ export class DemoMeetingApp implements
       return DefaultDeviceController.synthesizeVideoDevice('smpte');
     }
 
-    if (value === 'None') {
-      return null;
+    return value;
+  }
+
+  private videoFilterToProcessor(videoFilter: string): VideoFrameProcessor | null {
+    this.log(`Choosing video filter ${this.videoFilter}`);
+
+    if (videoFilter === 'Emojify') {
+      return new EmojifyVideoFrameProcessor('🚀');
     }
 
-    return value;
+    if (videoFilter === 'CircularCut') {
+      return new CircularCut();
+    }
+
+    if (videoFilter === 'NoOp') {
+      return new NoOpVideoFrameProcessor();
+    }
+
+    return null;
+  }
+
+  private chosenDevice: DefaultVideoTransformDevice;
+  private chosenProcessor: string;
+
+  private async stopChosenTransformDevice(): Promise<void> {
+    await this.chosenDevice?.stop();
+    this.chosenDevice = null;
+  }
+
+  private async videoInputSelectionWithOptionalFilter(innerDevice: Device): Promise<VideoInputDevice> {
+    if (this.videoFilter === 'None') {
+      return innerDevice;
+    }
+    
+    if (this.chosenDevice && this.videoFilter === this.chosenProcessor) {
+      this.chosenDevice = this.chosenDevice.chooseNewInnerDevice(innerDevice);
+      return this.chosenDevice;
+    }
+
+    const proc = this.videoFilterToProcessor(this.videoFilter);
+    this.chosenProcessor = this.videoFilter;
+    // different processor
+    if (this.chosenDevice) {
+      await this.chosenDevice.stop();
+    }
+
+    this.chosenDevice = new DefaultVideoTransformDevice(new ConsoleLogger('Demo'), innerDevice, [proc]);
```

Comment:
Try not to mix solidifying the selection of the name with the update of the device. Do one first and then the other.

Reply:
fixed.